diff --git a/LevelUP.csproj b/LevelUP.csproj
index 033b8c0..64de9fc 100644
--- a/LevelUP.csproj
+++ b/LevelUP.csproj
@@ -8,8 +8,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.NET" Version="3.11.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
+    <PackageReference Include="Discord.NET" Version="3.17.2" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
   </ItemGroup>
   
   <ItemGroup>
