diff --git a/lib/cmd-r b/lib/cmd-r
index 54a2b6c..429ab4e 100644
--- a/lib/cmd-r
+++ b/lib/cmd-r
@@ -1,2 +1,15 @@
 #!/bin/bash
+if [ ! "$(id -u)" == "$(id cmdr)" ] && [ "$(id -u)" != "0" ]; then
+    1>&2 echo CMD-R needs to run as its own user.
+    1>&2 echo Use sudo to start this script.
+    1>&2 echo
+    1>&2 echo Runuser will be used to start CMD-R as its own user.
+    exit 1
+fi
+
+if [ ! "$(id -u)" == "$(id cmdr)" ]; then
+    runuser --user cmdr -- cmd-r "$@"
+    exit $?
+fi
+
 /usr/lib/cmd-r/cmd-r
