diff --git a/lib/install-c.sh b/lib/install-c.sh
index 873b136..9a9e387 100644
--- a/lib/install-c.sh
+++ b/lib/install-c.sh
@@ -4,35 +4,44 @@ if [ ! -d "./build" ]; then
     exit
 fi
 
+libdir="."
+if [ "$1" != "" ]; then
+    libdir="$1"
+fi
+
 if [ ! -f "/usr/bin/cmd-r" ]; then
+    useradd -rmd /usr/lib/cmd-r cmdr &>/dev/null
+    usermod -L cmdr &>/dev/null
+
     echo Installing CMD-R...
     
     mkdir /usr/lib/cmd-r
     cp build/* /usr/lib/cmd-r -r -f
     
-    cp lib/cmd-r /usr/bin/cmd-r -f
-    cp lib/update-cmdr /usr/bin/update-cmdr -f
-    chmod 777 /usr/lib/cmd-r
+    cp $libdir/lib/cmd-r /usr/bin/cmd-r -f
+    cp $libdir/lib/update-cmdr /usr/bin/update-cmdr -f
+    cp $libdir/lib/cmdr.service /usr/lib/systemd/system/cmdr.service -f
+    chmod 700 /usr/lib/cmd-r
+    chown cmdr /usr/lib/cmd-r
     
     echo CMD-R has been installed, you can run it by running \'cmd-r\'
     echo Uninstall CMD-R by running \'make uninstall\' inside the downloaded repo.
 else
-    read -p "CMD-R is already installed, do you want to update the installation? [Y/n] " -n 1 -r
-    echo
-    if [[ $REPLY =~ ^[Yy]$ ]]
-    then
-        echo
-        echo Updating CMD-R...
+    useradd -rmd /usr/lib/cmd-r cmdr &>/dev/null
+    usermod -L cmdr &>/dev/null
     
-        cp build/* /usr/lib/cmd-r -r -f
+    echo Updating CMD-R...
+
+    cp build/* /usr/lib/cmd-r -r -f
+
+    cp $libdir/lib/cmd-r /usr/bin/cmd-r -f
+    cp $libdir/lib/update-cmdr /usr/bin/update-cmdr -f
+    cp $libdir/lib/cmdr.service /usr/lib/systemd/system/cmdr.service -f
+    chmod 700 /usr/lib/cmd-r
+    chown cmdr /usr/lib/cmd-r
     
-        cp lib/cmd-r /usr/bin/cmd-r -f
-        cp lib/update-cmdr /usr/bin/update-cmdr -f
-        chmod 777 /usr/lib/cmd-r
-        
-        echo Update completed, run CMD-R by running \'cmd-r\'
-        echo Uninstall CMD-R by running \'make uninstall\' inside the downloaded repo.
-        echo 
-        echo Tip: You can also use \'update-cmdr\' to install the latest update from the repo.
-    fi
+    echo Update completed, run CMD-R by running \'cmd-r\'
+    echo Uninstall CMD-R by running \'make uninstall\' inside the downloaded repo.
+    echo 
+    echo Tip: You can also use \'update-cmdr\' to install the latest update from the repo.
 fi
