diff --git a/lib/Update and start CMD-R.sh b/lib/Update and start CMD-R.sh
index 2f13044..07ea58a 100644
--- a/lib/Update and start CMD-R.sh	
+++ b/lib/Update and start CMD-R.sh	
@@ -1,5 +1,13 @@
 #!/bin/bash
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
-$DIR/update
-$DIR/cmd-r $@
+"$DIR/prestart/CMD-R" "$DIR" "Updating CMD-R..." &
+CMD_R_PID=$!
+"$DIR/update"
+echo 1 > "$DIR/exitprestart"
+while [ -d /proc/$CMD_R_PID ]
+do
+    sleep 1
+done
+rm "$DIR/exitprestart"
+"$DIR/cmd-r" $@
