diff --git a/CMD-R/HelpCmdModule/HelpCommand.cs b/CMD-R/HelpCmdModule/HelpCommand.cs
index edc6386..fc2cd62 100644
--- a/CMD-R/HelpCmdModule/HelpCommand.cs
+++ b/CMD-R/HelpCmdModule/HelpCommand.cs
@@ -5,10 +5,11 @@ using System.Threading.Tasks;
 using Discord;
 using Discord.WebSocket;
 using System.Threading;
+using Discord.Rest;
 
 namespace CMDR.HelpCmdModule
 {
-    public class HelpCommand : SystemCommand
+    public class HelpCommand : SystemCommand, DmSupportedCommand
     {
         public override string commandid => "help";
 
@@ -26,8 +27,8 @@ namespace CMDR.HelpCmdModule
 
         public CmdCategory c = new CmdCategory("noncategorized", "Commands that don't have a category");
         public override CmdCategory[] Categories => new CmdCategory[] { c };
-        
-        public override async Task OnExecuteFromDiscord(SocketGuild guild, SocketUser user, SocketTextChannel channel, SocketMessage messageobject, string fullmessage, string arguments_string, List<string> arguments)
+
+        public override async Task OnExecuteFromDiscord(SocketGuild guild, SocketGuildUser user, SocketTextChannel channel, SocketSlashCommand ev, string fullmessage, string arguments_string, List<string> arguments)
         {
             int page = 0;
             List<string> pages = new List<string>();
@@ -40,10 +41,13 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     bool found = false;
-                    foreach (CmdCategory cat in command.Categories) {
-                        if (cat.name.ToLower() == arguments[0].ToLower()) {
+                    foreach (CmdCategory cat in command.Categories)
+                    {
+                        if (cat.name.ToLower() == arguments[0].ToLower())
+                        {
                             found = true;
                             break;
                         }
@@ -58,10 +62,13 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     bool found = false;
-                    foreach (CmdCategory cat in command.Categories) {
-                        if (cat.name.ToLower().StartsWith(arguments[0].ToLower())) {
+                    foreach (CmdCategory cat in command.Categories)
+                    {
+                        if (cat.name.ToLower().StartsWith(arguments[0].ToLower()))
+                        {
                             found = true;
                             break;
                         }
@@ -76,10 +83,13 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     bool found = false;
-                    foreach (CmdCategory cat in command.Categories) {
-                        if (cat.name.ToLower().Contains(arguments[0].ToLower())) {
+                    foreach (CmdCategory cat in command.Categories)
+                    {
+                        if (cat.name.ToLower().Contains(arguments[0].ToLower()))
+                        {
                             found = true;
                             break;
                         }
@@ -94,7 +104,8 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     if (command.commandid.ToLower() != arguments[0].ToLower())
                         continue;
                 }
@@ -105,7 +116,8 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     if (!command.commandid.ToLower().StartsWith(arguments[0].ToLower()))
                         continue;
                 }
@@ -116,7 +128,8 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     if (!command.commandid.ToLower().Contains(arguments[0].ToLower()))
                         continue;
                 }
@@ -127,7 +140,8 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     if (!command.description.ToLower().StartsWith(arguments[0].ToLower()))
                         continue;
                 }
@@ -138,7 +152,8 @@ namespace CMDR.HelpCmdModule
 
             foreach (SystemCommand command in GetBot().commands)
             {
-                if (arguments.Count != 0) {
+                if (arguments.Count != 0)
+                {
                     if (!command.description.ToLower().Contains(arguments[0].ToLower()))
                         continue;
                 }
@@ -147,16 +162,20 @@ namespace CMDR.HelpCmdModule
                     commands.Add(command);
             }
 
-            foreach (SystemCommand command in commands) {
-                if (!Bot.GetBot().CheckPermissions(command.permissionnode, user, guild)) {
+            foreach (SystemCommand command in commands)
+            {
+                if (!Bot.GetBot().CheckPermissions(command.permissionnode, user, guild))
+                {
                     continue;
                 }
-                if (currentPage.Length + 3 >= 2000) {
+                string msg = ("\n - " + command.commandid + (command.helpsyntax == "" ? "" : " " + command.helpsyntax) + " - " + command.description);
+                if (currentPage.Length + msg.Length + 3 >= 4000)
+                {
                     currentPage += "```";
                     pages.Add(currentPage);
                     currentPage = "```A list of known commands (page " + pages.Count + "):";
                 }
-                currentPage += ("\n - " + command.commandid + (command.helpsyntax == "" ? "" : " " + command.helpsyntax) + " - " + command.description);    
+                currentPage += msg;
             }
             if (currentPage.Length != ("```A list of known commands (page " + pages.Count + "):").Length)
             {
@@ -164,9 +183,13 @@ namespace CMDR.HelpCmdModule
                 pages.Add(currentPage);
             }
 
-            Discord.Rest.RestUserMessage message = await channel.SendMessageAsync(pages[page]);
+            RestFollowupMessage message = await ev.FollowupAsync(embed: new EmbedBuilder()
+                    .WithTitle("Command listing and syntax")
+                    .WithColor(Color.Blue)
+                    .WithDescription(pages[page])
+                .Build());
 
-            Func<Cacheable<IUserMessage, ulong>, ISocketMessageChannel, SocketReaction, Task> handler = new Func<Cacheable<IUserMessage, ulong>, ISocketMessageChannel, SocketReaction, Task>((arg1, arg2, arg3) =>
+            Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task> handler = new Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task>((arg0, arg1, arg3) =>
             {
                 if (arg3.MessageId == message.Id && arg3.Channel.Id == message.Channel.Id)
                 {
@@ -191,8 +214,11 @@ namespace CMDR.HelpCmdModule
                             page--;
 
                             var oldmsg = message;
-                            message = channel.SendMessageAsync(pages[page]).GetAwaiter().GetResult();
-                            oldmsg.DeleteAsync().GetAwaiter().GetResult();
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command listing and syntax")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
 
                             if (page != 0)
                                 message.AddReactionAsync(em1).GetAwaiter().GetResult();
@@ -221,8 +247,11 @@ namespace CMDR.HelpCmdModule
                             page++;
 
                             var oldmsg = message;
-                            message = channel.SendMessageAsync(pages[page]).GetAwaiter().GetResult();
-                            oldmsg.DeleteAsync().GetAwaiter().GetResult();
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command listing and syntax")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
 
                             if (page != 0)
                                 message.AddReactionAsync(em1).GetAwaiter().GetResult();
@@ -234,24 +263,309 @@ namespace CMDR.HelpCmdModule
                 return null;
             });
 
-            Bot.GetBot().client.ReactionAdded += handler;
-            Bot.GetBot().client.MessageDeleted += (arg11, arg22) =>
+            new Thread(() =>
             {
-                if (arg11.Id == message.Id)
+                Bot.GetBot().client.ReactionAdded += handler;
+                Bot.GetBot().client.MessageDeleted += (arg11, arg22) =>
+                {
+                    if (arg11.Id == message.Id)
+                        Bot.GetBot().client.ReactionAdded -= handler;
+                    return null;
+                };
+
+                Task.Run(() =>
+                {
+                    Thread.Sleep(30000);
                     Bot.GetBot().client.ReactionAdded -= handler;
-                return null;
-            };
-            
-            if (pages.Count != 1)
-            {
-                if (page != pages.Count)
-                   await message.AddReactionAsync(em2);
-            }
+                });
+                
+                if (pages.Count != 1)
+                {
+                    if (page != pages.Count)
+                        message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                }
+            }).Start();
         }
 
         public override void OnExecuteFromTerminal(string fullcommand, string arguments_string, List<string> arguments)
         {
 
         }
+
+        public async Task OnExecuteFromDM(SocketUser user, SocketDMChannel channel, SocketSlashCommand ev, string fullmessage, string arguments_string, List<string> arguments)
+        {
+            int page = 0;
+            List<string> pages = new List<string>();
+
+            Emoji em1 = new Emoji("\u23EE");
+            Emoji em2 = new Emoji("\u23ED");
+
+            String currentPage = "```A list of known commands:";
+            List<SystemCommand> commands = new List<SystemCommand>();
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    bool found = false;
+                    foreach (CmdCategory cat in command.Categories)
+                    {
+                        if (cat.name.ToLower() == arguments[0].ToLower())
+                        {
+                            found = true;
+                            break;
+                        }
+                    }
+                    if (!found)
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    bool found = false;
+                    foreach (CmdCategory cat in command.Categories)
+                    {
+                        if (cat.name.ToLower().StartsWith(arguments[0].ToLower()))
+                        {
+                            found = true;
+                            break;
+                        }
+                    }
+                    if (!found)
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    bool found = false;
+                    foreach (CmdCategory cat in command.Categories)
+                    {
+                        if (cat.name.ToLower().Contains(arguments[0].ToLower()))
+                        {
+                            found = true;
+                            break;
+                        }
+                    }
+                    if (!found)
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    if (command.commandid.ToLower() != arguments[0].ToLower())
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    if (!command.commandid.ToLower().StartsWith(arguments[0].ToLower()))
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    if (!command.commandid.ToLower().Contains(arguments[0].ToLower()))
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    if (!command.description.ToLower().StartsWith(arguments[0].ToLower()))
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in GetBot().commands)
+            {
+                if (!(command is DmSupportedCommand))
+                    continue;
+                if (arguments.Count != 0)
+                {
+                    if (!command.description.ToLower().Contains(arguments[0].ToLower()))
+                        continue;
+                }
+
+                if (!commands.Contains(command))
+                    commands.Add(command);
+            }
+
+            foreach (SystemCommand command in commands)
+            {
+                if (!Bot.GetBot().CheckPermissions(command.permissionnode, user, null))
+                {
+                    continue;
+                }
+                string msg = ("\n - " + command.commandid + (command.helpsyntax == "" ? "" : " " + command.helpsyntax) + " - " + command.description);
+                if (currentPage.Length + msg.Length + 3 >= 2000)
+                {
+                    currentPage += "```";
+                    pages.Add(currentPage);
+                    currentPage = "```A list of known commands (page " + pages.Count + "):";
+                }
+                currentPage += msg;
+            }
+            if (currentPage.Length != ("```A list of known commands (page " + pages.Count + "):").Length)
+            {
+                currentPage += "```";
+                pages.Add(currentPage);
+            }
+
+
+            RestFollowupMessage message = await ev.FollowupAsync(embed: new EmbedBuilder()
+                    .WithTitle("Command listing and syntax")
+                    .WithColor(Color.Blue)
+                    .WithDescription(pages[page])
+                .Build());
+
+            Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task> handler = new Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task>((arg0, arg1, arg3) =>
+            {
+                if (arg3.MessageId == message.Id && arg3.Channel.Id == message.Channel.Id)
+                {
+                    if (arg3.Emote.Name == em1.Name)
+                    {
+                        bool changed = false;
+                        var v = message.GetReactionUsersAsync(em1, int.MaxValue).GetAsyncEnumerator();
+                        while (true)
+                        {
+                            if (v.Current != null)
+                                foreach (IUser usr in v.Current)
+                                    if (!usr.Id.Equals(Bot.GetBot().client.CurrentUser.Id))
+                                    {
+                                        message.RemoveReactionAsync(em1, usr).GetAwaiter().GetResult();
+                                        changed = true;
+                                    }
+                            if (!v.MoveNextAsync().GetAwaiter().GetResult())
+                                break;
+                        }
+                        if (changed)
+                        {
+                            page--;
+
+                            var oldmsg = message;
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command listing and syntax")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
+
+                            if (page != 0)
+                                message.AddReactionAsync(em1).GetAwaiter().GetResult();
+                            if (page != pages.Count - 1)
+                                message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                        }
+                    }
+                    if (arg3.Emote.Name == em2.Name)
+                    {
+                        bool changed = false;
+                        var v = message.GetReactionUsersAsync(em2, int.MaxValue).GetAsyncEnumerator();
+                        while (true)
+                        {
+                            if (v.Current != null)
+                                foreach (IUser usr in v.Current)
+                                    if (!usr.Id.Equals(Bot.GetBot().client.CurrentUser.Id))
+                                    {
+                                        message.RemoveReactionAsync(em1, usr).GetAwaiter().GetResult();
+                                        changed = true;
+                                    }
+                            if (!v.MoveNextAsync().GetAwaiter().GetResult())
+                                break;
+                        }
+                        if (changed)
+                        {
+                            page++;
+
+                            var oldmsg = message;
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command listing and syntax")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
+
+                            if (page != 0)
+                                message.AddReactionAsync(em1).GetAwaiter().GetResult();
+                            if (page != pages.Count - 1)
+                                message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                        }
+                    }
+                }
+                return null;
+            });
+
+            new Thread(() =>
+            {
+                Bot.GetBot().client.ReactionAdded += handler;
+                Bot.GetBot().client.MessageDeleted += (arg11, arg22) =>
+                {
+                    if (arg11.Id == message.Id)
+                        Bot.GetBot().client.ReactionAdded -= handler;
+                    return null;
+                };
+
+                Task.Run(() =>
+                {
+                    Thread.Sleep(30000);
+                    Bot.GetBot().client.ReactionAdded -= handler;
+                });
+                
+                if (pages.Count != 1)
+                {
+                    if (page != pages.Count)
+                        message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                }
+            }).Start();
+        }
     }
 }
