diff --git a/CMD-R/HelpCmdModule/LsCmdCategoriesCommand.cs b/CMD-R/HelpCmdModule/LsCmdCategoriesCommand.cs
index 4f73e73..3479f8d 100644
--- a/CMD-R/HelpCmdModule/LsCmdCategoriesCommand.cs
+++ b/CMD-R/HelpCmdModule/LsCmdCategoriesCommand.cs
@@ -1,12 +1,14 @@
 using System;
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 using Discord;
+using Discord.Rest;
 using Discord.WebSocket;
 
 namespace CMDR.HelpCmdModule
 {
-    public class LsCmdCategoriesCommand : SystemCommand
+    public class LsCmdCategoriesCommand : SystemCommand, DmSupportedCommand
     {
          public override string commandid => "lscategories";
 
@@ -25,7 +27,7 @@ namespace CMDR.HelpCmdModule
         public CmdCategory c = new CmdCategory("noncategorized", "Commands that don't have a category");
         public override CmdCategory[] Categories => new CmdCategory[] { c };
         
-        public override async Task OnExecuteFromDiscord(SocketGuild guild, SocketUser user, SocketTextChannel channel, SocketMessage messageobject, string fullmessage, string arguments_string, List<string> arguments)
+        public override async Task OnExecuteFromDiscord(SocketGuild guild, SocketGuildUser user, SocketTextChannel channel, SocketSlashCommand ev, string fullmessage, string arguments_string, List<string> arguments)
         {
             int page = 0;
             List<string> pages = new List<string>();
@@ -77,12 +79,13 @@ namespace CMDR.HelpCmdModule
             }
 
             foreach (CmdCategory cat in categories) {
-                if (currentPage.Length + 3 >= 2000) {
+                string msg = ("\n - " + cat.name + " - " + cat.description);
+                if (currentPage.Length + msg.Length + 3 >= 2000) {
                     currentPage += "```";
                     pages.Add(currentPage);
                     currentPage = "```A list of known command categories (page " + pages.Count + "):";
                 }
-                currentPage += ("\n - " + cat.name + " - " + cat.description);    
+                currentPage += msg;
             }
             if (currentPage.Length != ("```A list of known command categories (page " + pages.Count + "):").Length)
             {
@@ -90,9 +93,13 @@ namespace CMDR.HelpCmdModule
                 pages.Add(currentPage);
             }
 
-            Discord.Rest.RestUserMessage message = await channel.SendMessageAsync(pages[page]);
+            RestFollowupMessage message = await ev.FollowupAsync(embed: new EmbedBuilder()
+                    .WithTitle("Command category listing")
+                    .WithColor(Color.Blue)
+                    .WithDescription(pages[page])
+                .Build());
 
-            Func<Cacheable<IUserMessage, ulong>, ISocketMessageChannel, SocketReaction, Task> handler = new Func<Cacheable<IUserMessage, ulong>, ISocketMessageChannel, SocketReaction, Task>((arg1, arg2, arg3) =>
+            Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task> handler = new Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task>((arg0, arg1, arg3) =>
             {
                 if (arg3.MessageId == message.Id && arg3.Channel.Id == message.Channel.Id)
                 {
@@ -117,8 +124,11 @@ namespace CMDR.HelpCmdModule
                             page--;
 
                             var oldmsg = message;
-                            message = channel.SendMessageAsync(pages[page]).GetAwaiter().GetResult();
-                            oldmsg.DeleteAsync().GetAwaiter().GetResult();
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command category listing")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
 
                             if (page != 0)
                                 message.AddReactionAsync(em1).GetAwaiter().GetResult();
@@ -147,8 +157,11 @@ namespace CMDR.HelpCmdModule
                             page++;
 
                             var oldmsg = message;
-                            message = channel.SendMessageAsync(pages[page]).GetAwaiter().GetResult();
-                            oldmsg.DeleteAsync().GetAwaiter().GetResult();
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command category listing")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
 
                             if (page != 0)
                                 message.AddReactionAsync(em1).GetAwaiter().GetResult();
@@ -160,24 +173,203 @@ namespace CMDR.HelpCmdModule
                 return null;
             });
 
-            Bot.GetBot().client.ReactionAdded += handler;
-            Bot.GetBot().client.MessageDeleted += (arg11, arg22) =>
+            new Thread(() =>
             {
-                if (arg11.Id == message.Id)
+                Bot.GetBot().client.ReactionAdded += handler;
+                Bot.GetBot().client.MessageDeleted += (arg11, arg22) =>
+                {
+                    if (arg11.Id == message.Id)
+                        Bot.GetBot().client.ReactionAdded -= handler;
+                    return null;
+                };
+
+                Task.Run(() =>
+                {
+                    Thread.Sleep(30000);
                     Bot.GetBot().client.ReactionAdded -= handler;
-                return null;
-            };
-            
-            if (pages.Count != 1)
-            {
-                if (page != pages.Count)
-                   await message.AddReactionAsync(em2);
-            }
+                });
+                
+                if (pages.Count != 1)
+                {
+                    if (page != pages.Count)
+                        message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                }
+            }).Start();
         }
 
         public override void OnExecuteFromTerminal(string fullcommand, string arguments_string, List<string> arguments)
         {
 
         }
+
+        public async Task OnExecuteFromDM(SocketUser user, SocketDMChannel channel, SocketSlashCommand ev, string fullmessage, string arguments_string, List<string> arguments)
+        {
+            int page = 0;
+            List<string> pages = new List<string>();
+
+            Emoji em1 = new Emoji("\u23EE");
+            Emoji em2 = new Emoji("\u23ED");
+
+            String currentPage = "```A list of known command categories:";
+            List<CmdCategory> categories = new List<CmdCategory>();
+            foreach (CmdCategory cat in Bot.GetBot().CmdCategories) {
+                if (arguments.Count != 0) {
+                    if (cat.name.ToLower() != arguments[0].ToLower())
+                        continue;
+                }
+                if (!categories.Contains(cat))
+                    categories.Add(cat);
+            }
+            foreach (CmdCategory cat in Bot.GetBot().CmdCategories) {
+                if (arguments.Count != 0) {
+                    if (!cat.name.ToLower().StartsWith(arguments[0].ToLower()))
+                        continue;
+                }
+                if (!categories.Contains(cat))
+                    categories.Add(cat);
+            }
+            foreach (CmdCategory cat in Bot.GetBot().CmdCategories) {
+                if (arguments.Count != 0) {
+                    if (!cat.name.ToLower().Contains(arguments[0].ToLower()))
+                        continue;
+                }
+                if (!categories.Contains(cat))
+                    categories.Add(cat);
+            }
+            foreach (CmdCategory cat in Bot.GetBot().CmdCategories) {
+                if (arguments.Count != 0) {
+                    if (!cat.description.ToLower().StartsWith(arguments[0].ToLower()))
+                        continue;
+                }
+                if (!categories.Contains(cat))
+                    categories.Add(cat);
+            }
+            foreach (CmdCategory cat in Bot.GetBot().CmdCategories) {
+                if (arguments.Count != 0) {
+                    if (!cat.description.ToLower().Contains(arguments[0].ToLower()))
+                        continue;
+                }
+                if (!categories.Contains(cat))
+                    categories.Add(cat);
+            }
+
+            foreach (CmdCategory cat in categories) {
+                string msg = ("\n - " + cat.name + " - " + cat.description);
+                if (currentPage.Length + msg.Length + 3 >= 2000) {
+                    currentPage += "```";
+                    pages.Add(currentPage);
+                    currentPage = "```A list of known command categories (page " + pages.Count + "):";
+                }
+                currentPage += msg;
+            }
+            if (currentPage.Length != ("```A list of known command categories (page " + pages.Count + "):").Length)
+            {
+                currentPage += "```";
+                pages.Add(currentPage);
+            }
+
+            RestFollowupMessage message = await ev.FollowupAsync(embed: new EmbedBuilder()
+                    .WithTitle("Command category listing")
+                    .WithColor(Color.Blue)
+                    .WithDescription(pages[page])
+                .Build());
+
+            Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task> handler = new Func<Cacheable<IUserMessage, ulong>, Cacheable<IMessageChannel, ulong>, SocketReaction, Task>((arg0, arg1, arg3) =>
+            {
+                if (arg3.MessageId == message.Id && arg3.Channel.Id == message.Channel.Id)
+                {
+                    if (arg3.Emote.Name == em1.Name)
+                    {
+                        bool changed = false;
+                        var v = message.GetReactionUsersAsync(em1, int.MaxValue).GetAsyncEnumerator();
+                        while (true)
+                        {
+                            if (v.Current != null)
+                                foreach (IUser usr in v.Current)
+                                    if (!usr.Id.Equals(Bot.GetBot().client.CurrentUser.Id))
+                                    {
+                                        message.RemoveReactionAsync(em1, usr).GetAwaiter().GetResult();
+                                        changed = true;
+                                    }
+                            if (!v.MoveNextAsync().GetAwaiter().GetResult())
+                                break;
+                        }
+                        if (changed)
+                        {
+                            page--;
+
+                            var oldmsg = message;
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command category listing")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
+
+                            if (page != 0)
+                                message.AddReactionAsync(em1).GetAwaiter().GetResult();
+                            if (page != pages.Count - 1)
+                                message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                        }
+                    }
+                    if (arg3.Emote.Name == em2.Name)
+                    {
+                        bool changed = false;
+                        var v = message.GetReactionUsersAsync(em2, int.MaxValue).GetAsyncEnumerator();
+                        while (true)
+                        {
+                            if (v.Current != null)
+                                foreach (IUser usr in v.Current)
+                                    if (!usr.Id.Equals(Bot.GetBot().client.CurrentUser.Id))
+                                    {
+                                        message.RemoveReactionAsync(em1, usr).GetAwaiter().GetResult();
+                                        changed = true;
+                                    }
+                            if (!v.MoveNextAsync().GetAwaiter().GetResult())
+                                break;
+                        }
+                        if (changed)
+                        {
+                            page++;
+
+                            var oldmsg = message;
+                            message.ModifyAsync(t2 => t2.Embed = new EmbedBuilder()
+                                    .WithTitle("Command category listing")
+                                    .WithColor(Color.Blue)
+                                    .WithDescription(pages[page])
+                                .Build()).GetAwaiter().GetResult();
+
+                            if (page != 0)
+                                message.AddReactionAsync(em1).GetAwaiter().GetResult();
+                            if (page != pages.Count - 1)
+                                message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                        }
+                    }
+                }
+                return null;
+            });
+
+            new Thread(() =>
+            {
+                Bot.GetBot().client.ReactionAdded += handler;
+                Bot.GetBot().client.MessageDeleted += (arg11, arg22) =>
+                {
+                    if (arg11.Id == message.Id)
+                        Bot.GetBot().client.ReactionAdded -= handler;
+                    return null;
+                };
+
+                Task.Run(() =>
+                {
+                    Thread.Sleep(30000);
+                    Bot.GetBot().client.ReactionAdded -= handler;
+                });
+                
+                if (pages.Count != 1)
+                {
+                    if (page != pages.Count)
+                        message.AddReactionAsync(em2).GetAwaiter().GetResult();
+                }
+            }).Start();
+        }
     }
 }
\ No newline at end of file
