diff --git a/Rolling.csproj b/Rolling.csproj
index c36b366..964625e 100755
--- a/Rolling.csproj
+++ b/Rolling.csproj
@@ -8,8 +8,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.NET" Version="3.4.1" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="Discord.NET" Version="3.11.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
   </ItemGroup>
   
   <ItemGroup>
