diff --git a/Rolling.csproj b/Rolling.csproj
index 964625e..959980d 100755
--- a/Rolling.csproj
+++ b/Rolling.csproj
@@ -2,13 +2,13 @@
 
   <PropertyGroup>
     <ProjectName>Rolling</ProjectName>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <AssemblyName>Module-$(ProjectName)</AssemblyName>
     <TargetName>Module</TargetName>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.NET" Version="3.11.0" />
+    <PackageReference Include="Discord.NET" Version="3.12.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
   </ItemGroup>
   
