diff --git a/CMD-R/BotModule.cs b/CMD-R/BotModule.cs
index 60eef49..7ee67de 100644
--- a/CMD-R/BotModule.cs
+++ b/CMD-R/BotModule.cs
@@ -12,6 +12,10 @@ namespace CMDR
             return Bot.GetBot();
         }
 
+        public Server[] GetServers() {
+            return GetBot().servers.ToArray();
+        }
+
         public void RegisterCommand(SystemCommand command)
         {
             if (GetBot().CmdCategories.Find(t => t.name.ToLower() == command.commandid.ToLower()) != null)
@@ -61,7 +65,7 @@ namespace CMDR
 
         public void SaveConfig()
         {
-            File.WriteAllText(storagepath+"/config.xml", Serializer.Serialize(GetConfig()));
+            File.WriteAllText(storagepath + "/config.xml", Serializer.Serialize(GetConfig()));
         }
 
         public void LoadConfig()
