diff --git a/CMD-R/BotModule.cs b/CMD-R/BotModule.cs
index 60eef49..9a5d015 100644
--- a/CMD-R/BotModule.cs
+++ b/CMD-R/BotModule.cs
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using Discord.WebSocket;
 
 namespace CMDR
 {
@@ -12,29 +13,37 @@ namespace CMDR
             return Bot.GetBot();
         }
 
+        public Server[] GetServers()
+        {
+            return GetBot().servers.ToArray();
+        }
+
         public void RegisterCommand(SystemCommand command)
         {
+            if (GetBot().lockedCmdReg)
+                throw new InvalidOperationException("Legacy commands can only be registered during init due to limitations with discord command registration");
+
             if (GetBot().CmdCategories.Find(t => t.name.ToLower() == command.commandid.ToLower()) != null)
                 throw new Exception("Error: a category exists with that command id.");
 
             if (command.Categories.Any(t => t.name.ToLower() == command.commandid.ToLower()))
                 throw new Exception("Error: a category exists with that command id.");
-            
+
             if (command.commandid.Contains(" "))
                 throw new FormatException("Command id cannot contain space symbols.");
 
             Bot.WriteLine();
             Bot.WriteLine("Registering command " + command.commandid + " from module " + id + "...");
             Bot.WriteLine("    ID = " + command.commandid);
-            if (command.helpsyntax != "" ) Bot.WriteLine("    Syntax = " + command.helpsyntax);
+            if (command.helpsyntax != "") Bot.WriteLine("    Syntax = " + command.helpsyntax);
             Bot.WriteLine("    Description = " + command.description);
             Bot.WriteLine("    Permission node = " + command.permissionnode);
             Bot.WriteLine("    Allow call from discord = " + command.allowDiscord);
-            if (command.allowDiscord)  Bot.WriteLine("    Allow call from discord without command specifier = " + command.setNoCmdPrefix);
+            if (command.allowDiscord) Bot.WriteLine("    Allow call from discord without command specifier = " + command.setNoCmdPrefix);
             Bot.WriteLine("    Allow call from terminal = " + command.allowTerminal);
 
             string catPretty = "";
-            foreach (CmdCategory c in command.Categories) 
+            foreach (CmdCategory c in command.Categories)
             {
                 if (catPretty.Length != 0)
                     catPretty += ", ";
@@ -42,7 +51,7 @@ namespace CMDR
 
                 if (GetBot().CmdCategories.Find(t => t.name == c.name) == null)
                 {
-                    GetBot().CmdCategories.Add(c); 
+                    GetBot().CmdCategories.Add(c);
                 }
             }
 
@@ -61,7 +70,7 @@ namespace CMDR
 
         public void SaveConfig()
         {
-            File.WriteAllText(storagepath+"/config.xml", Serializer.Serialize(GetConfig()));
+            File.WriteAllText(storagepath + "/config.xml", Serializer.Serialize(GetConfig()));
         }
 
         public void LoadConfig()
@@ -76,6 +85,10 @@ namespace CMDR
         public abstract string id { get; }
         public abstract string moduledesctiption { get; }
 
+        public virtual void PrepareFramework(Bot bot) { }
+        public virtual void ConfigureClient(Bot bot, DiscordSocketClient client) { }
+        public virtual void ClientLogin(Bot bot, DiscordSocketClient client) { }
+
         public abstract void PreInit(Bot bot);
         public abstract void Init(Bot bot);
         public abstract void RegisterCommands(Bot bot);
