diff --git a/CMD-R/CMD-R.sln b/CMD-R/CMD-R.sln
new file mode 100644
index 0000000..3332291
--- /dev/null
+++ b/CMD-R/CMD-R.sln
@@ -0,0 +1,17 @@
+
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CMD-R", "CMD-R.csproj", "{B5BCAED2-F0AE-45F2-B938-8F0D32479722}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{B5BCAED2-F0AE-45F2-B938-8F0D32479722}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{B5BCAED2-F0AE-45F2-B938-8F0D32479722}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{B5BCAED2-F0AE-45F2-B938-8F0D32479722}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{B5BCAED2-F0AE-45F2-B938-8F0D32479722}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+EndGlobal
