diff --git a/CMD-R/DmSupportedCommand.cs b/CMD-R/DmSupportedCommand.cs
new file mode 100644
index 0000000..6c170e7
--- /dev/null
+++ b/CMD-R/DmSupportedCommand.cs
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+
+namespace CMDR
+{
+    public interface DmSupportedCommand
+    {
+        public Task OnExecuteFromDM(SocketUser user, SocketDMChannel channel, SocketMessage messageobject, string fullmessage, string arguments_string, List<string> arguments);
+
+    }
+}
\ No newline at end of file
