diff --git a/CMD-R/PermChecker.cs b/CMD-R/PermChecker.cs
index fa2a644..626ef29 100644
--- a/CMD-R/PermChecker.cs
+++ b/CMD-R/PermChecker.cs
@@ -12,6 +12,11 @@ namespace CMDR
             PermissionsAllowed = new List<string>(DefaultPermissions);
             PermissionsBlacklist = new List<string>();
 
+            if  (server == null){
+                PermissionsAllowed.Add("sys.anyone");
+                return;
+            }
+            
             if (server.OwnerId == usr.Id)
             {
                 PermissionsAllowed.Add("*");
@@ -21,6 +26,12 @@ namespace CMDR
             List<SocketRole> roles = new List<SocketRole>(server.Roles).FindAll(t => t.Members.Contains(user));
             foreach (Role role in servers.Find(t => t.id == server.Id).roles.FindAll(t => roles.Find(t2 => t2.Id == t.roleid) != null))
             {
+                SocketRole r = server.GetRole(role.roleid);
+                if (r != null) {
+                    if (r.Permissions.Administrator) {
+                        PermissionsAllowed.Add("*");
+                    }
+                }
                 foreach (string perm in role.permissions)
                 {
                     if (!PermissionsAllowed.Contains(perm))
