diff --git a/CMD-R/Program.cs b/CMD-R/Program.cs
index fc7bc02..8374080 100644
--- a/CMD-R/Program.cs
+++ b/CMD-R/Program.cs
@@ -9,11 +9,15 @@ using System.Threading.Tasks;
 using Discord;
 using Discord.WebSocket;
 using System.IO.Compression;
+using System.Numerics;
+using Discord.Rest;
 
 namespace CMDR
 {
     public partial class Bot
     {
+        internal bool lockedCmdReg = false;
+
         public static void WriteLine(string msg = "")
         {
             Terminal.WriteLine(msg);
@@ -28,6 +32,7 @@ namespace CMDR
         static Bot _bot;
         public List<Server> servers;
 
+        internal List<GatewayIntents> intents = new List<GatewayIntents>();
         public List<BotModule> modules = new List<BotModule>();
         public List<SystemCommand> commands = new List<SystemCommand>();
         public List<CmdCategory> CmdCategories = new List<CmdCategory>();
@@ -35,7 +40,6 @@ namespace CMDR
         {
             return _bot;
         }
-        public string prefix = "+";
         public string token = "";
         public string path;
         public ulong guild = 0;
@@ -109,9 +113,9 @@ namespace CMDR
             Console.ForegroundColor = b;
 
             path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
-            Bot.WriteLine("Booting CMD-R...");
-            Bot.WriteLine("AerialWorks CMD-R is Free(Libre) and Open Source Software (FLOSS),\nand will always be licensed under the GPL-2.0 license.");
-            Bot.WriteLine("------------------------------------------------------------------");
+            Bot.WriteLine("Starting CMD-R (Ultimate Commander Version)...");
+            Bot.WriteLine("AerialWorks Ultimate Commander is Free(Libre) and Open Source Software (FLOSS),\nand will always be licensed under the GPL-2.0 license.");
+            Bot.WriteLine("-------------------------------------------------------------------------------");
             Bot.WriteLine();
             Bot.WriteLine("System path: " + path);
             Directory.CreateDirectory(path + "/Modules");
@@ -131,11 +135,16 @@ namespace CMDR
                 newToken = true;
             }
 
-            Console.CancelKeyPress += new ConsoleCancelEventHandler(delegate (Object sender, ConsoleCancelEventArgs e) {
-                if (e.SpecialKey == ConsoleSpecialKey.ControlBreak) {
-                    Bot.WriteLine("Shutting down CMD-R, CTRL+BREAK detected...");
+            Console.CancelKeyPress += new ConsoleCancelEventHandler(delegate (Object sender, ConsoleCancelEventArgs e)
+            {
+                if (e.SpecialKey == ConsoleSpecialKey.ControlBreak)
+                {
+                    Bot.WriteLine("Shutting down Ultimate Commander, CTRL+BREAK detected...");
                     client.SetGameAsync("Shutting down...").GetAwaiter().GetResult();
+                    Server.RunSaveAll(true);
+
                     client.SetStatusAsync(UserStatus.Invisible).GetAwaiter().GetResult();
+                    client.SetGameAsync("").GetAwaiter().GetResult();
                     client.StopAsync().GetAwaiter().GetResult();
                     client.Dispose();
                     Server.RunSaveAll(true);
@@ -144,18 +153,52 @@ namespace CMDR
                 }
                 e.Cancel = true;
             });
+
+            // Configure client
             Bot.WriteLine("Starting Discord.NET framework...");
-            client = new DiscordSocketClient();
+
+            // Load itnent config
+            if (!File.Exists(path + "/Intents.cfg"))
+                File.WriteAllText(path + "/Intents.cfg", "# Add required intents here\n# Each intent should be on its own line\n# If none are given, only unprivileged intents are used\n# See Discord.NET GatewayIntents enum for names\n\n");
+            foreach (string line in File.ReadAllLines(path + "/Intents.cfg"))
+            {
+                if (line.StartsWith("#") || line == "")
+                    continue;
+
+                // Parse
+                try
+                {
+                    AddRequiredIntent(Enum.Parse<GatewayIntents>(line));
+                }
+                catch
+                {
+                    throw new FormatException("Unknown intent: " + line + " in Intents.cfg");
+                }
+            }
+            
+            // Create intent list
+            GatewayIntents intentI = GatewayIntents.AllUnprivileged;
+            foreach (GatewayIntents i in intents)
+                intentI = intentI | i;
+
+            // Apply intents
+            client = new DiscordSocketClient(new DiscordSocketConfig()
+            {
+                GatewayIntents = intentI
+            });
 
             await client.LoginAsync(TokenType.Bot, token);
             await client.StartAsync();
-            await client.SetGameAsync("Starting CMD-R...");
+            await client.SetGameAsync("Starting application...");
 
-            if (newToken) {
+            if (newToken)
+            {
                 Bot.WriteLine("\nSaved to file Bot.cfg, edit it to change the token.");
                 File.WriteAllText(path + "/Bot.cfg", token);
             }
 
+            Bot.WriteLine("Waiting for servers to load... Please note that the bot needs to be in at least ONE server for it to load.");
+            Bot.WriteLine("Please confirm all required intents are properly configured in the dev portal, as otherwise the bot can hang.");
             while (client.ConnectionState == ConnectionState.Connecting || client.ConnectionState == ConnectionState.Disconnected) { Thread.Sleep(100); }
             while (client.Guilds.Count == 0 || client.Guilds.FirstOrDefault().Name == null || client.Guilds.FirstOrDefault().Name == "") { Thread.Sleep(100); }
 
@@ -209,30 +252,31 @@ namespace CMDR
             {
                 loadPackage(info);
             }
-            
+
             // Load CMF (Cyan Modfile) formatted packages
             // CMF are zips too, just like jars
             //
-            // By allowing CMFs to be loaded in  CMD-R, one can create a one-fits-all package
+            // By allowing CMFs to be loaded in Ultimate Commander, one can create a one-fits-all package
             // When you create a Connective module, you can add a pointer file, dll, and mod.manifest.ccfg document
-            // to make it compatible with CMD-R, Connective and Cyan. (if you combine the packages)
+            // to make it compatible with Ultimate Commander, Connective and Cyan. (if you combine the packages)
             foreach (FileInfo info in new DirectoryInfo(path + "/Module Packages").GetFiles("*.cmf"))
             {
                 loadPackage(info);
             }
-            
-            void loadPackage(FileInfo info) {
+
+            void loadPackage(FileInfo info)
+            {
                 bool update = false;
-                
+
                 if (!Directory.Exists(path + "/Module Packages/" + Path.GetFileNameWithoutExtension(info.FullName))) update = true; // Make sure the package gets extracted if new
                 else
                 {
                     // Check if package archive is newer than extracted version
                     string oldpatch = (File.Exists(path + "/Module Packages/" + Path.GetFileNameWithoutExtension(info.FullName) + "/patch.ver") ? File.ReadAllText(path + "/Module Packages/" + Path.GetFileNameWithoutExtension(info.FullName) + "/patch.ver") : "");
                     // If the patch.ver file does not exist, the string is empty
-                    
+
                     string newpatch = "";
-                    
+
                     // Read the archive for getting the new version file, skip if patch.ver is not in the archive
                     ZipArchive file = ZipFile.OpenRead(path + "/Module Packages/" + Path.GetFileName(info.FullName));
                     if (file.Entries.ToList().Find(t => t.Name == "patch.ver") != null)
@@ -245,20 +289,20 @@ namespace CMDR
                         }
                         catch
                         {
-                            
+
                         }
                     }
-                    
+
                     int newv = 0;
                     int oldv = 0;
-                    
+
                     // Parse if the strings are not empty
                     if (newpatch != "") newv = int.Parse(newpatch);
                     if (oldpatch != "") oldv = int.Parse(oldpatch);
-                    
+
                     update = (oldv < newv); // Check if the archive is newer than the folder
                 }
-                
+
                 // Extract files if new/updated
                 if (update)
                 {
@@ -293,21 +337,21 @@ namespace CMDR
                             b = Console.ForegroundColor;
                             Console.ForegroundColor = ConsoleColor.Red;
                             Bot.WriteLine("CRITICAL ERROR: Unable to restore or save storage folder, one already exists in both temp and install output!");
-                            Bot.WriteLine("DO NOT RESTART CMD-R BEFORE FIXING THIS ERROR OR FILES WILL GET LOST!");
+                            Bot.WriteLine("DO NOT RESTART ULTIMATE COMMANDER BEFORE FIXING THIS ERROR OR FILES WILL GET LOST!");
                             Console.ForegroundColor = b;
-                            throw new FieldAccessException("Unable to move storage back to origin, already exists, consider CMD-R crashed.");
+                            throw new FieldAccessException("Unable to move storage back to origin, already exists, consider Ultimate Commander crashed.");
                         }
                         if (Directory.Exists(path + "/temp/Storage"))
                         {
                             Directory.Move(path + "/temp/Storage", path + "/Module Packages/" + Path.GetFileNameWithoutExtension(info.FullName) + "/Storage");
                         }
-                        
-                        #pragma warning disable CA2200
+
+#pragma warning disable CA2200
                         throw e;
-                        #pragma warning restore CA2200
+#pragma warning restore CA2200
                     }
                 }
-                
+
                 ConfigDictionary<String, String> conf = new ConfigDictionary<string, string>();
                 if (File.Exists(path + "/Module Packages/" + Path.GetFileNameWithoutExtension(info.FullName) + "/pointer.targets"))
                 {
@@ -394,22 +438,44 @@ namespace CMDR
             Bot.WriteLine("Set the autosave interval to " + Server.AutoSaveMinutes + " minutes.");
             Server.StartAutoSaveThread();
 
+            // Legacy to modern discord
+            Bot.WriteLine();
+            Bot.WriteLine("Initializing support for legacy commands...");
+            lockedCmdReg = true;
+
+            // Bind event
+            client.JoinedGuild += InitLegacyCommands;
+            await InitLegacyDmCommands(client);
+
+            // Go through existing
+            foreach (SocketGuild guild in client.Guilds)
+            {
+                await InitLegacyCommands(guild);
+            }
+
+            // Bind handler
+            client.SlashCommandExecuted += OnCommand;
+
+            // Init modern commands
+            // FIXME
+
             Bot.WriteLine();
             Bot.WriteLine("Execute quit or exit to close.");
 
-            client.MessageReceived += onMessage;
             await client.SetGameAsync("");
             while (true)
             {
-                string command = Terminal.ReadLine().Replace("\0", "");
+                string command = Terminal.ReadLine("CMD-R: $<path>").Replace("\0", "");
                 if (command == "quit" || command == "exit")
                 {
                     await client.SetGameAsync("Shutting down...");
+                    Server.RunSaveAll(true);
 
                     await client.SetStatusAsync(UserStatus.Invisible);
+                    await client.SetGameAsync("");
                     await client.StopAsync();
                     client.Dispose();
-                    Server.RunSaveAll(true);
+                    Server.RunSaveAll(true); // just in case
                     Server.StopAutoSaveThread();
                     Environment.Exit(0);
                 }
@@ -604,11 +670,239 @@ namespace CMDR
             }
         }
 
+        private async Task InitLegacyCommands(SocketGuild guild)
+        {
+            // Guild loaded
+
+            // De-register old commands
+            bool commandAvailable = false;
+            IReadOnlyCollection<SocketApplicationCommand> arr = null;
+            try
+            {
+                arr = await guild.GetApplicationCommandsAsync();
+                commandAvailable = true;
+                Bot.WriteLine("De-registering old commands for Guild " + guild.Name + "...");
+                foreach (SocketApplicationCommand command in arr)
+                {
+                    if (command.Description.StartsWith("[LEGACY] ") && command.Description.EndsWith(" (Legacy CMD-R Command)"))
+                    {
+                        string name = command.Name;
+                        if (!commands.Any(t => t.commandid == name && t.allowDiscord))
+                        {
+                            try
+                            {
+                                // Old command
+                                Bot.WriteLine("Deleting command " + name + " in Guild " + guild.Name + "...");
+                                await command.DeleteAsync();
+                            }
+                            catch
+                            {
+                            }
+                        }
+                    }
+                }
+            }
+            catch
+            {
+            }
+
+            // Register new commands
+            if (commandAvailable)
+            {
+                Bot.WriteLine("Registering missing commands for Guild " + guild.Name + "...");
+
+                // Find commands to register
+                List<SystemCommand> reregister = new List<SystemCommand>();
+                foreach (SystemCommand command in commands)
+                {
+                    if (command.allowDiscord && !arr.Any(t =>
+                    {
+                        if (t.Description.StartsWith("[LEGACY] ") && t.Description.EndsWith(" (Legacy CMD-R Command)") && t.Name == command.commandid)
+                        {
+                            // Check description
+                            if (("[LEGACY] " + command.description + " (Legacy CMD-R Command)").Length > 100)
+                            {
+                                if (t.Description == ("[LEGACY] Legacy command " + command.commandid + " (Legacy CMD-R Command)"))
+                                {
+                                    return true;
+                                }
+                            }
+                            else
+                            {
+                                if (t.Description == "[LEGACY] " + command.description + " (Legacy CMD-R Command)")
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                        return false;
+                    }))
+                    {
+                        // Add
+                        Bot.WriteLine("Command to register: " + command.commandid + " in Guild " + guild.Name);
+                        reregister.Add(command);
+                    }
+                }
+
+                // Prepare to add commands
+                Bot.WriteLine("Registering commands for Guild " + guild.Name + "...");
+                foreach (SystemCommand command in reregister)
+                {
+                    // Create command entry
+                    Bot.WriteLine("Registering command: " + command.commandid + " in Guild " + guild.Name + "...");
+                    SlashCommandBuilder b = new SlashCommandBuilder();
+                    b.WithName(command.commandid);
+                    if (("[LEGACY] " + command.description + " (Legacy CMD-R Command)").Length > 100)
+                    {
+                        b.WithDescription("[LEGACY] Legacy command " + command.commandid + " (Legacy CMD-R Command)");
+                    }
+                    else
+                    {
+                        b.WithDescription("[LEGACY] " + command.description + " (Legacy CMD-R Command)");
+                    }
+                    b.AddOption("arguments", ApplicationCommandOptionType.String, "Arguments to pass to this legacy command", false);
+                    try
+                    {
+                        await guild.CreateApplicationCommandAsync(b.Build());
+                    }
+                    catch
+                    {
+                    }
+                }
+
+                // Remove file
+                if (File.Exists(Bot.GetBot().path + "/Server Configs/" + guild.Id + "/commandsfailedregister"))
+                {
+                    File.Delete(Bot.GetBot().path + "/Server Configs/" + guild.Id + "/commandsfailedregister");
+                }
+            }
+            else
+            {
+                // Error
+                // DM the server operator if not done yet
+                if (!File.Exists(Bot.GetBot().path + "/Server Configs/" + guild.Id + "/commandsfailedregister"))
+                {
+                    File.WriteAllText(Bot.GetBot().path + "/Server Configs/" + guild.Id + "/commandsfailedregister", "");
+                    try
+                    {
+                        // DM
+                        await (await client.GetUserAsync(guild.OwnerId)).SendMessageAsync(embed: new EmbedBuilder()
+                            .WithTitle("Missing permissions")
+                            .WithDescription(client.CurrentUser.Username + " is missing required permissions!\n\nPlease re-add the " + client.CurrentUser.Username + " bot with the `Create Application Commands` permission, you can do so by re-inviting with an updated invite link.")
+                            .WithColor(Color.Red)
+                        .Build());
+                    }
+                    catch
+                    {
+                    }
+                }
+            }
+        }
+
+        private async Task InitLegacyDmCommands(DiscordSocketClient client)
+        {// De-register old commands
+            bool commandAvailable = false;
+            IReadOnlyCollection<SocketApplicationCommand> arr = null;
+            try
+            {
+                arr = await client.GetGlobalApplicationCommandsAsync();
+                commandAvailable = true;
+                Bot.WriteLine("De-registering old global commands...");
+                foreach (SocketApplicationCommand command in arr)
+                {
+                    if (command.Description.StartsWith("[LEGACY] ") && command.Description.EndsWith(" (Legacy CMD-R Command)"))
+                    {
+                        string name = command.Name;
+                        if (!commands.Any(t => t.commandid == name && t.allowDiscord && t is DmSupportedCommand))
+                        {
+                            try
+                            {
+                                // Old command
+                                Bot.WriteLine("Deleting command " + name + "...");
+                                await command.DeleteAsync();
+                            }
+                            catch
+                            {
+                            }
+                        }
+                    }
+                }
+            }
+            catch
+            {
+            }
+
+            // Register new commands
+            if (commandAvailable)
+            {
+                Bot.WriteLine("Registering missing global commands...");
+
+                // Find commands to register
+                List<SystemCommand> reregister = new List<SystemCommand>();
+                foreach (SystemCommand command in commands)
+                {
+                    if (command.allowDiscord && command is DmSupportedCommand && !arr.Any(t =>
+                    {
+                        if (t.Description.StartsWith("[LEGACY] ") && t.Description.EndsWith(" (Legacy CMD-R Command)") && t.Name == command.commandid)
+                        {
+                            // Check description
+                            if (("[LEGACY] " + command.description + " (Legacy CMD-R Command)").Length > 100)
+                            {
+                                if (t.Description == ("[LEGACY] Legacy command " + command.commandid + " (Legacy CMD-R Command)"))
+                                {
+                                    return true;
+                                }
+                            }
+                            else
+                            {
+                                if (t.Description == "[LEGACY] " + command.description + " (Legacy CMD-R Command)")
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                        return false;
+                    }))
+                    {
+                        // Add
+                        Bot.WriteLine("Command to register: " + command.commandid);
+                        reregister.Add(command);
+                    }
+                }
+
+                // Prepare to add commands
+                Bot.WriteLine("Registering global commands...");
+                foreach (SystemCommand command in reregister)
+                {
+                    // Create command entry
+                    Bot.WriteLine("Registering global command: " + command.commandid + "...");
+                    SlashCommandBuilder b = new SlashCommandBuilder();
+                    b.WithName(command.commandid);
+                    if (("[LEGACY] " + command.description + " (Legacy CMD-R Command)").Length > 100)
+                    {
+                        b.WithDescription("[LEGACY] Legacy command " + command.commandid + " (Legacy CMD-R Command)");
+                    }
+                    else
+                    {
+                        b.WithDescription("[LEGACY] " + command.description + " (Legacy CMD-R Command)");
+                    }
+                    b.AddOption("arguments", ApplicationCommandOptionType.String, "Arguments to pass to this legacy command", false);
+                    try
+                    {
+                        await client.CreateGlobalApplicationCommandAsync(b.Build());
+                    }
+                    catch
+                    {
+                    }
+                }
+            }
+        }
+
         public static void LoadBotModule(string filename, string filefolder, string classname = "{auto}", string namespacepath = "{auto}")
         {
             if (classname == "{auto}") classname = filename;
             if (namespacepath == "{auto}") namespacepath = filename;
-            Bot.WriteLine("Loading module file: " + filefolder + Path.DirectorySeparatorChar + filename + ".dll");
+            Bot.WriteLine("Loading module file: " + filefolder + Path.DirectorySeparatorChar + filename + "-" + namespacepath + "." + classname + ".dll");
             AppDomain currentDomain = AppDomain.CurrentDomain;
             currentDomain.AssemblyResolve += new ResolveEventHandler(LoadFromSameFolder);
 
@@ -620,14 +914,18 @@ namespace CMDR
                 Assembly assembly = Assembly.LoadFrom(assemblyPath);
                 return assembly;
             }
-            Assembly asm = Assembly.LoadFrom(filefolder + Path.DirectorySeparatorChar + filename + ".dll");
+
+            if (File.Exists(filefolder + Path.DirectorySeparatorChar + filename + ".dll"))
+                File.Move(filefolder + Path.DirectorySeparatorChar + filename + ".dll", filefolder + Path.DirectorySeparatorChar + filename + "-" + namespacepath + "." + classname + ".dll", true);
+
+            Assembly asm = Assembly.Load(File.ReadAllBytes(filefolder + Path.DirectorySeparatorChar + filename + "-" + namespacepath + "." + classname + ".dll"));
             Type type = asm.GetType(namespacepath + "." + classname);
 
             var script = Activator.CreateInstance(type) as BotModule;
 
             if (script == null)
             {
-                Bot.WriteLine("Module file: '" + filefolder + Path.DirectorySeparatorChar + filename + ".dll' does not have a class named " + classname + " in a namespace named " + namespacepath + ", cannot load it, skipping...");
+                Bot.WriteLine("Module file: '" + filefolder + Path.DirectorySeparatorChar + filename + "-" + namespacepath + "." + classname + ".dll' does not have a class named " + classname + " in a namespace named " + namespacepath + ", cannot load it, skipping...");
             }
             else
             {
@@ -635,13 +933,13 @@ namespace CMDR
                 Bot.WriteLine("Loading module: " + script.id + "...");
                 Bot.WriteLine("Module description: " + script.moduledesctiption);
                 Bot.GetBot().modules.Add(script);
-                Directory.CreateDirectory(Bot.GetBot().path + "/Modules/" + script.id + "/Storage");
-                if (!File.Exists(Bot.GetBot().path + "/Modules/" + script.id + "/Storage/config.xml") && File.Exists(Bot.GetBot().path + "/Modules/" + script.id + "/config-defaults.xml"))
+                Directory.CreateDirectory(filefolder + "/Storage");
+                if (!File.Exists(filefolder + "/Storage/config.xml") && File.Exists(filefolder + "/config-defaults.xml"))
                 {
-                    File.Copy(Bot.GetBot().path + "/Modules/" + script.id + "/config-defaults.xml", Bot.GetBot().path + "/Modules/" + script.id + "/Storage/config.xml");
+                    File.Copy(filefolder + "/config-defaults.xml", filefolder + "/Storage/config.xml");
                 }
                 script.modulepath = Path.GetFullPath(filefolder + Path.DirectorySeparatorChar + filename + ".dll");
-                script.storagepath = Path.GetFullPath(Bot.GetBot().path + Path.DirectorySeparatorChar + "Modules" + Path.DirectorySeparatorChar + script.id + Path.DirectorySeparatorChar + "Storage");
+                script.storagepath = Path.GetFullPath(filefolder + Path.DirectorySeparatorChar + "Storage");
                 script.LoadConfig();
                 Bot.WriteLine("Pre-initializing module: " + script.id + "...");
                 script.PreInit(Bot.GetBot());
@@ -650,6 +948,30 @@ namespace CMDR
             }
         }
 
+        public BotModule GetModule(string module)
+        {
+            foreach (BotModule mod in modules)
+            {
+                if (mod.id == module)
+                {
+                    return mod;
+                }
+            }
+            return null;
+        }
+
+        public bool IsModuleLoaded(string module)
+        {
+            foreach (BotModule mod in modules)
+            {
+                if (mod.id == module)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
 #pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
         private async Task Client_RoleDeleted(SocketRole arg)
 #pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
@@ -819,75 +1141,110 @@ namespace CMDR
             }
         }
 
-#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
-        private async Task onMessage(SocketMessage msg)
-#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        private async Task OnCommand(SocketSlashCommand ev)
         {
-            var message = msg as SocketUserMessage;
-            if (message == null) return;
-            SocketTextChannel ch = message.Channel as SocketTextChannel;
-            if (msg.Content.StartsWith(prefix, StringComparison.CurrentCulture) && ch != null && !msg.Content.StartsWith(prefix + " ", StringComparison.CurrentCulture) && msg.Content != prefix)
+            SocketUser user = ev.User;
+            if (ev.IsDMInteraction)
             {
-                bool found = false;
-                foreach (SystemCommand cmd in commands)
+                // Global command
+                if (ev.Channel is SocketDMChannel)
                 {
-                    if (cmd.allowDiscord)
-                    {
-                        string fullcmd = message.Content;
-                        string cmdid = fullcmd.Substring(prefix.Length);
-                        string arguments = "";
-                        if (fullcmd.Contains(" "))
-                        {
-                            arguments = cmdid.Substring(cmdid.IndexOf(" ", StringComparison.CurrentCulture) + 1);
-                            cmdid = cmdid.Remove(cmdid.IndexOf(" ", StringComparison.CurrentCulture));
-                        }
+                    // Get channel
+                    SocketDMChannel ch = ev.Channel as SocketDMChannel;
 
-                        if (cmd.commandid.ToLower() == cmdid.ToLower())
-                        {
-                            found = true;
+                    // Defer
+                    await ev.DeferAsync();
 
-                            if (CheckPermissions(cmd.permissionnode, msg.Author, ch.Guild))
+                    // Find command
+                    string name = ev.CommandName;
+                    SystemCommand cmd = commands.FirstOrDefault(t => t.commandid == name && t.allowDiscord && t is DmSupportedCommand);
+                    if (cmd != null)
+                    {
+                        if (cmd.permissionnode == "sys.anyone")
+                        {
+                            // Run
+                            string args = "";
+                            SocketSlashCommandDataOption opt = ev.Data.Options.FirstOrDefault();
+                            string fullCmd = cmd.commandid + (args != "" ? " " : "") + args;
+                            if (opt != null)
+                                args = opt.Value.ToString();
+                            try
                             {
-                                await cmd.OnExecuteFromDiscord(ch.Guild, msg.Author, ch, msg, fullcmd, arguments, GetArgumentListFromString(arguments));
+                                // Run
+                                await ((DmSupportedCommand)cmd).OnExecuteFromDM(user, ch, ev, fullCmd, args, GetArgumentListFromString(args));
                             }
-                            else
+                            finally
                             {
-                                await message.Channel.SendMessageAsync("```diff\n- I am sorry, but you are not allowed to run that command```");
+                                // Delete message if untouched
+                                RestInteractionMessage m = await ev.GetOriginalResponseAsync();
+                                if (m.Flags.HasValue && m.Flags.Value.HasFlag(MessageFlags.Loading))
+                                {
+                                    // Delete
+                                    await m.DeleteAsync();
+                                }
                             }
                         }
+                        else
+                        {
+                            await ev.RespondAsync(embed: new EmbedBuilder()
+                                .WithTitle("Insufficient Permissions")
+                                .WithDescription("Sorry, but you do not have the required permissions to use this command.")
+                                .WithColor(Color.Red)
+                            .Build());
+                        }
                     }
                 }
-                if (!found)
-                {
-                    await message.Channel.SendMessageAsync("I am sorry, but i don't recognize that command, use " + prefix + "help for commands.");
-                }
             }
-            else if (message.Channel is SocketDMChannel && message.Author.Id != client.CurrentUser.Id) await msg.Channel.SendMessageAsync("```diff\n- I am sorry, but CMD-R do not support direct messages yet.\n- Please go to a server and run " + prefix + "help for a list of commands```");
-            else {
-                foreach (SystemCommand cmd in commands)
+            else
+            {
+                // Guild command
+                if (ev.Channel is SocketTextChannel)
                 {
-                    if (cmd.allowDiscord && cmd.setNoCmdPrefix)
-                    {
-                        string fullcmd = message.Content;
-                        string cmdid = fullcmd;
-                        string arguments = "";
-                        if (fullcmd.Contains(" "))
-                        {
-                            arguments = cmdid.Substring(cmdid.IndexOf(" ", StringComparison.CurrentCulture) + 1);
-                            cmdid = cmdid.Remove(cmdid.IndexOf(" ", StringComparison.CurrentCulture));
-                        }
+                    // Get channels
+                    SocketTextChannel ch = ev.Channel as SocketTextChannel;
+                    SocketGuild guild = ch.Guild;
+
+                    // Defer
+                    await ev.DeferAsync();
 
-                        if (cmd.commandid.ToLower() == cmdid.ToLower())
+                    // Find command
+                    string name = ev.CommandName;
+                    SystemCommand cmd = commands.FirstOrDefault(t => t.commandid == name && t.allowDiscord);
+                    if (cmd != null)
+                    {
+                        // FIXME: better permission system
+                        if (CheckPermissions(cmd.permissionnode, user, ch.Guild))
                         {
-                            if (CheckPermissions(cmd.permissionnode, msg.Author, ch.Guild))
+                            // Run
+                            string args = "";
+                            SocketSlashCommandDataOption opt = ev.Data.Options.FirstOrDefault();
+                            string fullCmd = cmd.commandid + (args != "" ? " " : "") + args;
+                            if (opt != null)
+                                args = opt.Value.ToString();
+                            try
                             {
-                                await cmd.OnExecuteFromDiscord(ch.Guild, msg.Author, ch, msg, fullcmd, arguments, GetArgumentListFromString(arguments));
+                                // Run
+                                await cmd.OnExecuteFromDiscord(guild, user as SocketGuildUser, ch, ev, fullCmd, args, GetArgumentListFromString(args));
                             }
-                            else
+                            finally
                             {
-                                await message.Channel.SendMessageAsync("```diff\n- I am sorry, but you are not allowed to run that command```");
+                                // Delete message if untouched
+                                RestInteractionMessage m = await ev.GetOriginalResponseAsync();
+                                if (m.Flags.HasValue && m.Flags.Value.HasFlag(MessageFlags.Loading))
+                                {
+                                    // Delete
+                                    await m.DeleteAsync();
+                                }
                             }
                         }
+                        else
+                        {
+                            await ev.RespondAsync(embed: new EmbedBuilder()
+                                .WithTitle("Insufficient Permissions")
+                                .WithDescription("Sorry, but you do not have the required permissions to use this command.")
+                                .WithColor(Color.Red)
+                            .Build());
+                        }
                     }
                 }
             }
@@ -904,5 +1261,11 @@ namespace CMDR
                 await t.GetAwaiter().GetResult().DeleteAsync();
             });
         }
+
+        public void AddRequiredIntent(GatewayIntents intent)
+        {
+            if (!intents.Contains(intent))
+                intents.Add(intent);
+        }
     }
 }
