diff --git a/configure b/configure
index dbd3215..80c17f6 100644
--- a/configure
+++ b/configure
@@ -49,10 +49,10 @@ echo Checking dependencies...
 missingdepends=false
 
 requiredvermsbuild=15.6.0.0
-requiredvernuget=5.6.0.6591
-requiredvermake=4.3
+requiredvernuget=5.6.0.6489
+requiredvermake=4.2
 requiredvergit=2.28.0
-requiredverdotnet=5.0.205
+requiredverdotnet=7.0.0
 
 echo Searching for dotnet...
 if command -v dotnet &> /dev/null 
