diff --git a/ConnectiveHTTP Module (Ultimate Commander)/gradle/wrapper/gradle-wrapper.properties b/ConnectiveHTTP Module (Ultimate Commander)/gradle/wrapper/gradle-wrapper.properties
new file mode 100644
index 0000000..00e33ed
--- /dev/null
+++ b/ConnectiveHTTP Module (Ultimate Commander)/gradle/wrapper/gradle-wrapper.properties	
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.1-bin.zip
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
