diff --git a/Link-r.csproj b/Link-r.csproj
index fc2b9c6..2a44b21 100755
--- a/Link-r.csproj
+++ b/Link-r.csproj
@@ -9,8 +9,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.NET" Version="3.4.1" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="Discord.NET" Version="3.11.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
   </ItemGroup>
   
   <ItemGroup>
