diff --git a/PermissionManager.csproj b/PermissionManager.csproj
index 6b973aa..fa7595c 100644
--- a/PermissionManager.csproj
+++ b/PermissionManager.csproj
@@ -8,8 +8,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.NET" Version="3.11.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
+    <PackageReference Include="Discord.NET" Version="3.17.2" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
   </ItemGroup>
   
   <ItemGroup>
