diff --git a/Crossover.csproj b/Crossover.csproj
index 2f5f488..894d0de 100644
--- a/Crossover.csproj
+++ b/Crossover.csproj
@@ -2,14 +2,14 @@
 
   <PropertyGroup>
     <ProjectName>Crossover</ProjectName>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <AssemblyName>Module-$(ProjectName)</AssemblyName>
     <TargetName>Module</TargetName>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.NET" Version="3.11.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
+    <PackageReference Include="Discord.NET" Version="3.12.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
   </ItemGroup>
   
   <ItemGroup>
@@ -24,7 +24,7 @@
    <PropertyGroup Condition=" '$(RunConfiguration)' == 'Default' ">
     <StartAction>Program</StartAction>
     <StartProgram>${ProjectDir}\run\CMD-R.exe</StartProgram>
-    <StartArguments>--enable-debug ASMLD:{${ProjectDir}\bin\Debug\net6.0\Module.dll} --disable-debug</StartArguments>
+    <StartArguments>--enable-debug ASMLD:{${ProjectDir}\bin\Debug\net8.0\Module.dll} --disable-debug</StartArguments>
     <StartWorkingDirectory>${ProjectDir}\run</StartWorkingDirectory>
     <ExternalConsole>true</ExternalConsole>
   </PropertyGroup>
